synapseclient/client.py

```diff
@@ -2211,6 +2212,27 @@ def getProvenance(self, entity, version=None):
             uri = "/entity/%s/generatedBy" % id_of(entity)
         return Activity(data=self.restGET(uri))
 
+    @deprecated.sphinx.deprecated(
```

Comment:
Great point.

You're right - as of right now this will get the latest activity (by default) with the option to specify a version in the args. I like your thought of this `get_provenance` evolving to return a list of activities - either all, or a paginated list of activities.

Reply:
👍🏼 one thing we have to be weary of is that there is also `setProvenance`.  It's not ideal that the web client has this for editing an activity
<img width="890" alt="image" src="https://github.com/Sage-Bionetworks/synapsePythonClient/assets/10556275/87297de6-6722-4a59-8211-8625bb030935">

The inconsistency is a bit frustrating.

Reply:
I am going to deprecate all of these `provenance` methods then to use the `activity` verbiage and also swap to snake_case

Reply:
Just a note, users _may_ be irritated with this change as what they're used to is "provenance".  If thats the case, we can just re-introduce `get` and `set` provenance, but as said above, we have the opportunity to think about just exactly what it means to set and get provenance.